omnigibson/robots/stretch.py

```diff
@@ -0,0 +1,203 @@
+            ["link_head_nav_cam", "link_head_pan"],
+            ["link_mast", "link_lift"],
+            ["link_lift", "link_aruco_shoulder"],
+            ["link_lift", "link_arm_l4"],
+            ["link_lift", "link_arm_l3"],
+            ["link_lift", "link_arm_l2"],
+            ["link_lift", "link_arm_l1"],
+            ["link_arm_l4", "link_arm_l3"],
+            ["link_arm_l4", "link_arm_l2"],
+            ["link_arm_l4", "link_arm_l1"],
+            ["link_arm_l3", "link_arm_l2"],
+            ["link_arm_l3", "link_arm_l1"],
+            ["link_arm_l2", "link_arm_l1"],
+            ["link_arm_l0", "link_arm_l1"],
+            ["link_arm_l0", "link_arm_l2"],
+            ["link_arm_l0", "link_arm_l3"],
+            ["link_arm_l0", "link_arm_l4"],
+            ["link_arm_l0", "link_arm_l1"],
+            ["link_arm_l0", "link_aruco_inner_wrist"],
+            ["link_arm_l0", "link_aruco_top_wrist"],
+            ["link_arm_l0", "link_wrist_yaw"],
+            ["link_arm_l0", "link_wrist_yaw_bottom"],
+            ["link_arm_l0", "link_wrist_pitch"],
+            ["link_wrist_yaw_bottom", "link_wrist_pitch"],
+            ["gripper_camera_link", "link_gripper_s3_body"],
+            ["link_gripper_s3_body", "link_aruco_d405"],
+            ["link_gripper_s3_body", "link_gripper_finger_left"],
+            ["link_gripper_finger_left", "link_aruco_fingertip_left"],
+            ["link_gripper_finger_left", "link_gripper_fingertip_left"],
+            ["link_gripper_s3_body", "link_gripper_finger_right"],
+            ["link_gripper_finger_right", "link_aruco_fingertip_right"],
+            ["link_gripper_finger_right", "link_gripper_fingertip_right"],
+            ["respeaker_base", "link_head"],
+            ["respeaker_base", "link_mast"],
+        ]
+
+    @property
+    def base_joint_names(self):
+        return ["joint_left_wheel", "joint_right_wheel"]
+
+    @property
+    def camera_joint_names(self):
+        return ["joint_head_pan", "joint_head_tilt"]
+
+    @property
+    def arm_link_names(self):
+        return {
+            self.default_arm: [
+                "link_mast",
+                "link_lift",
+                "link_arm_l4",
+                "link_arm_l3",
+                "link_arm_l2",
+                "link_arm_l1",
+                "link_arm_l0",
+                "link_aruco_inner_wrist",
+                "link_aruco_top_wrist",
+                "link_wrist_yaw",
+                "link_wrist_yaw_bottom",
+                "link_wrist_pitch",
+                "link_wrist_roll",
+            ]
+        }
+
+    @property
+    def arm_joint_names(self):
+        return {
+            self.default_arm: [
+                "joint_lift",
+                "joint_arm_l3",
+                "joint_arm_l2",
+                "joint_arm_l1",
+                "joint_arm_l0",
+                "joint_wrist_yaw",
+                "joint_wrist_pitch",
+                "joint_wrist_roll",
+            ]
```

Comment:
Why are there more arm links than arm joints? Is it because some of the joints are fixed joints?

Reply:
Yes, exactly!